apis/Google.Devtools.Aspnet/Google.Devtools.Aspnet/CloudErrorReportingExceptionLogger.cs

```diff
@@ -0,0 +1,182 @@
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="CloudErrorReportingExceptionLogger"/> using credentials as
+        /// defined by <see cref="GoogleCredential.GetApplicationDefaultAsync"/>.
+        /// </summary>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static CloudErrorReportingExceptionLogger Create(string projectId, string serviceName, string version)
+        {
+            return new CloudErrorReportingExceptionLogger(
+                ReportErrorsServiceClient.Create(), projectId, serviceName, version);
+        }
+
+        private CloudErrorReportingExceptionLogger(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version) : base()
+        {
+            _client = GaxPreconditions.CheckNotNull(client, nameof(client));
+            _projectId = ReportErrorsServiceClient.FormatProjectName(
+                GaxPreconditions.CheckNotNull(projectId, nameof(projectId)));
+            _serviceContext = new ServiceContext
+            {
+                Service = GaxPreconditions.CheckNotNull(serviceName, nameof(serviceName)),
+                Version = GaxPreconditions.CheckNotNull(version, nameof(version)),
+            };
+        }
+
+        public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
+        {
+            ReportedErrorEvent errorEvent = CreateReportRequest(context);
+            return _client.ReportErrorEventAsync(_projectId, errorEvent);
+        }
+
+        public override void Log(ExceptionLoggerContext context)
+        {
+            ReportedErrorEvent errorEvent = CreateReportRequest(context);
+            _client.ReportErrorEvent(_projectId, errorEvent);
+        }
+
+        public override bool ShouldLog(ExceptionLoggerContext context)
+        {
+            return context?.Exception != null;
+        }
+
+        /// <summary>
+        /// Gets information about the HTTP request and response when the exception occured 
+        /// and populates a <see cref="HttpRequestContext"/> object.
+        /// </summary>
+        private HttpRequestContext CreateHttpRequestContext(ExceptionLoggerContext exceptionLoggerContext)
+        {
+            ExceptionContext exceptionContext = exceptionLoggerContext.ExceptionContext;
+            HttpRequestMessage requestMessage = exceptionContext?.Request;
+            HttpResponseMessage responseMessage = exceptionContext?.Response;
+
+            HttpRequestContext requestContext = new HttpRequestContext()
+            {
+                Method = requestMessage?.Method?.ToString(),
```

Comment:
This is going to assign null to a string property, which isn't allowed in protos. Just put `?? ""` at the end and it should be fine. Ditto all the other properties.